migration/migrator/migrations/system/20211121152144_autograding_stacktrace_chgrp.py

```diff
@@ -0,0 +1,26 @@
+import os
+from pathlib import Path
+
+"""Migration for the Submitty system."""
+
+
+def up(config):
+    """
+    Run up migration.
+
+    :param config: Object holding configuration details about Submitty
+    :type config: migrator.config.Config
+    """
+    os.system("chgrp -R {} {}".format(config.submitty_users['daemonphp_group'], str(Path(config.submitty['submitty_data_dir'], 'logs', 'autograding_stack_traces'))))
+    os.system("chgrp -R {} {}".format(config.submitty_users['daemonphp_group'], str(Path(config.submitty['submitty_data_dir'], 'logs', 'autograding'))))
+
+
+def down(config):
+    """
+    Run down migration (rollback).
+
+    :param config: Object holding configuration details about Submitty
+    :type config: migrator.config.Config
+    """
+    os.system("chgrp -R {} {}".format(config.submitty_users['course_builders_group'], str(Path(config.submitty['submitty_data_dir'], 'logs', 'autograding_stack_traces'))))
+    os.system("chgrp -R {} {}".format(config.submitty_users['course_builders_group'], str(Path(config.submitty['submitty_data_dir'], 'logs', 'autograding'))))
```

Comment:
You don't need to use a migration to change the group of these directories. Just change the `chown` lines for these directories in `.setup/INSTALL_SUBMITTY_HELPER.sh`.

However, this does change the group from `course_builders` to `daemonphp_group`  meaning instructors would no longer be able to access these directories?

Reply:
Yes, the idea was that the instructors will use the autograding status page to view such information since ssh-ing into the submitty machine isn't a popular practice among instructors anyway. Changing this to `daemonphp_group` will allow the php script to read the logs which was not allowed when the group was `course_builders`.